### DISCUSSION TITLE
Number of steps

### DISCUSSION THREAD

**Discussion by hieu12a10cdag** [SWivid/F5-TTS#234] [Q&A] [answered]
Hi everyone, 
My dataset has 50k+ samples, and the batch size type is frame
Can somebody please explain to me why if I increase the Batch Size per GPU, then the number of steps per epoch will increase too?
I.e if I set Batch Size is 32, then each epoch I only have around 150 steps, but if batch size is 512, each epoch have more than 20k+ steps

**Comment by SWivid** [comment, marked as answer]
As the batch size type is `frame` rather than `sample`
Thus setting a batchsize of 32 meaning using a batch with maximum 32\*256/24000 = 0.34 seconds audio
512\*256/24000 = 5.46 seconds audio.  (256 is hop_length, 24000 is for 24khz sampling rate)
The batch size is not the number of audios `samples`.

**Reply by hieu12a10cdag** [reply]
Thank you, so how can I calculate the optimal batch size for my dataset?

**Reply by SWivid** [reply]
the optimal batchsize basically depends on your gpu memory (we just try squeeze as much into it first)
then you could refer to #57 pioneers' experiences, with gradient_accumulation to adjust

**Reply by hieu12a10cdag** [reply]
thank you very much! Btw in your opinion, if I want to train another language (In my case, Vietnamese TTS) then should I fine tune from current checkpoint or pre train from scratch?

**Reply by SWivid** [reply]
Both possible. If you have acceptable gpu resources and a certain amount of data to do training from scratch, it's better.

**Reply by hieu12a10cdag** [reply]
Hi, I do wonder why 64 batch size (frame type) make training an epoch much faster than 40k batch size? Would you mind to explain to me?

**Reply by SWivid** [reply]
you are just dropping all samples longer than 64 frames, you have not even trained on them!

**Reply by hieu12a10cdag** [reply]
So I just need to set the batch size equal to the longest sample in my dataset. Is that necessary to increase it over that point?
